### PR TITLE
#163303067 create meetup  api endpoint

### PR DESCRIPTION
**Pull request by JosephNjuguna**
### What does this PR do??
this pr enables a user to create a meetup
### Description of task to be completed??
- add create meetup api endpoint
- add a test for my meetup api endpoint
- update response message  as per LFA feedback
### How should this be tested?
clone this repo to your machine.
open the repo folder in terminal
add virtual env  `python3 -m virtualenv env`
export the APP environment ` export FLASK_ENV=development`
start the app by running following command `flask run`
open postman and navigate to following link `http://127.0.0.1:5000/api/v2/meetup`
test the endpoint with this data:
```
{ 
    "location":"Kenya ,Safaricom",
    "topic":"ViceChat",
    "happeningon":"1/1/2018",
    "tag":"@comeall",
    "isAdmin":"true"
}
```
### Relevat Pivotal tracker stories?
[pivotal tracker link ](https://www.pivotaltracker.com/story/show/163303067)
### Any background context??
i have added test for my create meetup api endpoint
### SCREENSHOTS AVAILABLE >
screenshot from 2019-01-23 14-37-23

screenshot from 2019-01-23 15-57-03

### CODE REVIEW

File: app/api/v2/models/meetup.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+"""
+#inbuilt modules
+import datetime
+#import uuid
+# local imports
+from app.api.v2.models.database import init_db
+from app.api.v2.models.basemodel import BaseModel
+#class
+class MeetUp(BaseModel):
+    def __init__(self, createdon="1/1/2018", location="unknown", topic="unknown", happeningon="1/1/2018", tags="unknown", user_state=""):
```
Comment (oagutu) [inline app/api/v2/models/meetup.py:12]: Why are there default values for `creation` and `happeningon`?
Also it's not advisable to have this many arguments passed to a function, you could pass them as `**kwargs`

File: app/api/v2/models/meetup.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        curr = self.db.cursor()
+        query = "SELECT isAdmin FROM users WHERE isAdmin = '%s'" % (user_state)
+        curr.execute(query)
+        return curr.fetchone() is not None
+
+    def create_meetup(self):
+        meetup_data = {
+            "topic": self.topic,
+        }
+        """check if question exist"""
```
Comment (oagutu) [inline app/api/v2/models/meetup.py:40]: I don't think this is supposed to be here

File: app/api/v2/views/meetupviews.py [hunk 3]
```diff
@@ -1,9 +1,65 @@
+import datetime
+#downloaded dependencies
 from flask_restful import Resource
+from flask import request, make_response, jsonify
+#local imports
+from app.api.v2.models.meetup import MeetUp
+from app.api.v2.utilis.validations import CheckData
+args = ""
+empty_loction ={"message":"Empty Meetup location"}
```
Comment (oagutu) [inline app/api/v2/views/meetupviews.py:9]: Typo: `location`

File: app/api/v2/views/meetupviews.py [hunk 4]
```diff
@@ -1,9 +1,65 @@
...
+            """pass meetup dict to Meetup Models"""
+            meetup_db = MeetUp(**meetupdata)
+            saved = meetup_db.create_meetup()
+            resp = {
+                "message": "Meetup Succesfully Created",
+                "meetup": topic,
+                "meetup_id": "{}".format(saved)
+            }
+            if saved == True:
+                return make_response(jsonify({"message":"Meet already up exist"}),409)
```
Comment (oagutu) [inline app/api/v2/views/meetupviews.py:59]: This should be, `Meetup already  exists`

**Review by oagutu — COMMENTED** [review]
- Omitting some fields from the request data causes the app to crash eg, `happeningon`
    - You don’t need `isAdmin` field  in the request data when creating a meetup
    - No validation on `happeningon` field value; this should not be the case; ie. check for valid date format
    - `tag` should be a list of strings.
    - Add validation for datatype to avoid app crashing when integer provided instead of string, eg for `happeningon` or `topic`
    - Why are the newly created meetup record details not included in the response?
    - Your tests for `create meetup` are still failing,  kindly look into this.

**Comment by JosephNjuguna** [comment]
thanks for the feedback. i will work on the feedback before our stand up meeting

**Pull request merged**